### PR TITLE
feat: provide props for ariaLabel - < Toast />, < AlertBanner />, < Combobox /> Fixes: #1642

### PR DESCRIPTION
**Pull request by viditagrawal56**
#### Description
-Fixes: #1642
-Provided AriaLabel props to override corresponding AriaLabel for i18n purposes
-No new Components were added.

#### Basic

- [X] Used plop (`npm run plop`) to create a new component.
- [X] PR has description.
- [X] New component is functional and uses Hooks.
- [X] Component is written in TypeScript.

#### Style

- [X] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [X] Component uses CSS Modules.
- [X] Design is compatible with [Monday Design System](https://design.monday.com/).
- [X] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-style) tokens
- [X] Displays correctly in all the themes

#### Storybook

- [X] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.stories.mdx` file.
- [X] Stories include all flows of using the component.
- [X] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [X] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

#### Accessibility

- [X] Component is accessible.
- [X] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--page)

### CODE REVIEW

File: src/components/AlertBanner/AlertBanner.tsx [hunk 1]
```diff
@@ -122,7 +122,7 @@ const AlertBanner: VibeComponent<AlertBannerProps> & {
               size={Button.sizes.SMALL}
               kind={Button.kinds.TERTIARY}
               color={isDarkBackground ? Button.colors.ON_INVERTED_BACKGROUND : Button.colors.ON_PRIMARY_COLOR}
-              ariaLabel="close-toast"
+              closeButtonAriaLabel="Close"
```
Comment (SergeyRoyt) [inline src/components/AlertBanner/AlertBanner.tsx:125]: You got it a bit wrong. Here it should be `ariaLabel={closeButtonAriaLabel}` and `closeButtonAriaLabel` should be a prop of `AlertBanner` with default value = `Close`. This way users will be able to pass other ariaLabel into the component e.g. when localization is needed and user is using the system in, let's say, German

File: src/components/Button/Button.tsx
Comment (SergeyRoyt) [inline src/components/Button/Button.tsx:1]: Please, revert these changes in Button component

File: src/components/Combobox/Combobox.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ export interface ComboboxProps extends VibeComponentProps {
    * Divider between categories sections
    */
   withCategoriesDivider?: boolean;
-  size?: typeof BASE_SIZES[keyof typeof BASE_SIZES];
+  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
```
Comment (SergeyRoyt) [inline src/components/Combobox/Combobox.tsx:46]: Doesn't pass lint

File: src/components/Search/Search.tsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ export interface SearchProps extends VibeComponentProps {
   setRef?: () => void;
   autoComplete?: string;
   /* BASE_SIZES is exposed on the component itself */
-  size?: typeof BASE_SIZES[keyof typeof BASE_SIZES];
+  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
```
Comment (SergeyRoyt) [inline src/components/Search/Search.tsx:30]: Doesn't pass lint

File: src/components/Combobox/Combobox.tsx [hunk 4]
```diff
@@ -284,7 +284,7 @@ const Combobox: React.FC<ComboboxProps> & {
             value={filterValue}
             wrapperClassName={cx(styles.comboboxSearchWrapper, searchWrapperClassName)}
             className={styles.comboboxSearch}
-            inputAriaLabel="Search for content"
+            searchInputAriaLabel="Search for content"
```
Comment (SergeyRoyt) [inline src/components/Combobox/Combobox.tsx:287]: Same here. It should be `inputAriaLabel={searchInputAriaLabel}` and searchInputAriaLabel should be a prop of Combobox with default value = `Search for content`

File: src/components/Search/Search.tsx
Comment (SergeyRoyt) [inline src/components/Search/Search.tsx:1]: Please, revert changes on this component

File: src/components/TextField/TextField.tsx
Comment (SergeyRoyt) [inline src/components/TextField/TextField.tsx:1]: Should be reverted

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Let's fix the notes I wrote you 🙏

File: src/components/Toast/Toast.tsx [hunk 5]
```diff
@@ -150,7 +150,7 @@ const Toast: FC<ToastProps> & { types?: typeof ToastType; actionTypes?: typeof T
             size={Button.sizes.SMALL}
             kind={Button.kinds.TERTIARY}
             color={Button.colors.ON_PRIMARY_COLOR}
-            ariaLabel="close-toast"
+            closeButtonAriaLabel="Close"
```
Comment (SergeyRoyt) [inline src/components/Toast/Toast.tsx:153]: Same here

**Comment by viditagrawal56** [comment]
@SergeyRoyt  Sorry for the obvious mistakes 😬
Will fix this ASAP

File: src/components/Button/Button.tsx
Comment (viditagrawal56) [reply src/components/Button/Button.tsx:1]: done!

File: src/components/Combobox/Combobox.tsx (hunk 2 above)
Comment (viditagrawal56) [reply src/components/Combobox/Combobox.tsx]: Fixed!

File: src/components/Search/Search.tsx (hunk 3 above)
Comment (viditagrawal56) [reply src/components/Search/Search.tsx]: Fixed!

File: src/components/Combobox/Combobox.tsx (hunk 4 above)
Comment (viditagrawal56) [reply src/components/Combobox/Combobox.tsx]: Fixed!

File: src/components/TextField/TextField.tsx
Comment (viditagrawal56) [reply src/components/TextField/TextField.tsx:1]: Done!

File: src/components/Toast/Toast.tsx (hunk 5 above)
Comment (viditagrawal56) [reply src/components/Toast/Toast.tsx]: Fixed!

**Comment by viditagrawal56** [comment]
@SergeyRoyt Hey! Thanks for the review and helping me fix the silly mistakes I made.
I am a beginner so I am learning a lot!
Let me know if you need any more changes 😄

File: src/components/Search/Search.tsx
Comment (viditagrawal56) [reply src/components/Search/Search.tsx:1]: Done!

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Well done! 😉
Now the only thing left is to update and push snapshot tests - we are using them to track the changes in our components - you just need to run `npm run test:update`, check it and commit and push
I also see that one jest test which depends on the previous value of aria-label is now failing - it's not the best practice to use aria-label to get the element in jest, so I propose to replace it with retrieving element by it's data-testid, in this case you should refer to `ComponentDefaultTestId.Button`

**Comment by viditagrawal56** [comment]
Hey @SergeyRoyt, I am a bit confused as to how to change toast-tests.jest.js
From what I understand from ComponentDefaultTestId.Button, the test ids are present inside the test/constants.ts inside the ComponentDefaultTestId Enum.
So I added the ComponentDefaultTestId.TOAST_BUTTON accordingly as seen in the screenshot below.

image

And I changed the toast-test.jest.js file accordingly as seen in the screenshot below.

image

I ran npm run test:update and all the test cases passed.
Is my implemntation correct or do I need to change anything let me know so I can commit my changes 🙏

**Comment by viditagrawal56** [comment]
> Hey, @viditagrawal56 ! It's close, but not a perfect solution, cause we are currently using `ComponentDefaultTestId.TOAST_BUTTON` inside `ToastButton` component - button which you can see on the screenshot image
> 
> So in your solution two elements would have same data-testid, which in general should be avoided
> 
> And in this close-button we currently don't pass any data-testid specifically - so it just gets Button default data-testid, that's why I told you to use `ComponentDefaultTestId.BUTTON`
> 
> But indeed the more correct solution would be to provide it a separate data-testid - it might break some existing tests for somebody using the library, but we don't care about it this much, as it only tests. So let's add a new value to `ComponentDefaultTestId` and call it `TOAST_CLOSE_BUTTON`, then let's pass it inside this close Button and use it in the tests to retrieve it

Sure! Thanks for the reply. I'll implement it ASAP.

**Comment by SergeyRoyt** [comment]
Hey, @viditagrawal56 !
It looks amazing, thanks!! 👑
I'd only ask you to resolve the merge conflicts and fix lint errors - for lint you can just run `npm run fix-lint` 🙏

**Comment by viditagrawal56** [comment]
Hey @SergeyRoyt,
I am so sorry for bothering you again but I just wanted to see if I am doing something wrong.
I am a beginner and this is the first time I am solving a merge conflict.
I did git pull in vs code so that I can see the conflicts in the vs code editor itself.
Fixed the conflicts and ran npm run fix-lint.
But now vs code shows so many staged changes 😨

image

Am I doing something wrong? Please guide me 🙏

**Comment by SergeyRoyt** [comment]
> Hey @SergeyRoyt, I am so sorry for bothering you again but I just wanted to see if I am doing something wrong. I am a beginner and this is the first time I am solving a merge conflict. I did git pull in vs code so that I can see the conflicts in the vs code editor itself. Fixed the conflicts and ran npm run fix-lint. But now vs code shows so many staged changes 😨
> 
> image
> 
> Am I doing something wrong? Please guide me 🙏

Hey @viditagrawal56 !
It seems like you doing it right so far. You just need to solve conflicts in two first files:
- For Toast file open merge conflict editor and solve it line by line
- For toast snapshots just take one of the versions and then run `npm run test:update` - it will just generate relevant snapshots
Than u'll need to stage both of the files
Regarding the other changes - it should be fine, you should have just pulled the latest master and then merge. So just commit and push them with the solved conflict files
Then check in this PR changes if u haven't accidentally bring any unrelated changes

**Comment by viditagrawal56** [comment]
@SergeyRoyt Ok will give it a try!
Thanks for helping me 😄

**Comment by viditagrawal56** [comment]
@SergeyRoyt made the commit. 
Did I mess up something?

**Comment by viditagrawal56** [comment]
I think I made a huge mistake.
I should have just staged the two conflicting files that I fixed and then commit.

**Comment by SergeyRoyt** [comment]
> I think I made a huge mistake. I should have just staged the two conflicting files that I fixed and then commit. Should i revert the commit?

@viditagrawal56 
No worries, man! All is fixable :) 
You can just hard reset to the latest commit before merge and try again
This time don't run lint on all of the files - that's the problem, my bad, sorry 😝Just configure `prettier` on save and prettify files you've been working on
Regarding the rest - it should be exactly how you did it
- fix lint issues in your files with prettier
- pull master & merge it to your branch
- solve Toast conflicts
- update snapshots to fix toast.snapshots conflicts
- stage everything, commit and **force** push
Then check again if you see correct files in this PR

**Comment by viditagrawal56** [comment]
Hey @SergeyRoyt Thanks for helping me!
I think everything is right this time.
Do I need to change anything?

**Comment by viditagrawal56** [comment]
All checks passed! 😀

**Review by SergeyRoyt — APPROVED** [review]
You are a champ, @viditagrawal56!!
Congratulations with the first contribution to Vibe! 🥳
Thanks a lot for your determination to complete the task! 👑
Keep up the good work, keep coding and improving! 💪

**Comment by viditagrawal56** [comment]
> You are a champ, @viditagrawal56!! Congratulations with the first contribution to Vibe! 🥳 Thanks a lot for your determination to complete the task! 👑 Keep up the good work, keep coding and improving! 💪

@SergeyRoyt A huge thanks to you for all the guidance you provided me throughout this task. 🙌
This was my biggest contribution yet and you helped me a lot, Thank you so much!
I love this community and would definitely try my best to contribute to it! 😄
